tratto/src/main/java/star/tratto/util/javaparser/JavaParserUtils.java

```diff
@@ -471,52 +656,189 @@ public static String getMethodSignature(MethodDeclaration methodDeclaration) {
+        List<String> exceptionList = getMethodUsageExceptions(methodUsage);
+        return (methodModifiers + " " + (typeParameterList.isEmpty() ? "" : "<" + String.join(", ", typeParameterList) + ">") +
+                " " + getTypeWithoutPackages(methodDeclaration.getReturnType()) +
+                " " + methodDeclaration.getName() +
+                "(" + String.join(", ", parameterList) + ")" +
+                (exceptionList.isEmpty() ? "" : " throws " + String.join(", ", exceptionList)))
+                .replaceAll(" +", " ").trim();
+    }
+
+    /**
+     * Gets the package of a given compilation unit.
+     *
+     * @param cu a compilation unit
+     * @return the package of the compilation unit
+     * @throws PackageDeclarationNotFoundException if the package declaration
+     * cannot be found
+     */
+    public static PackageDeclaration getPackageDeclarationFromCompilationUnit(
+            CompilationUnit cu
+    ) throws PackageDeclarationNotFoundException {
+        Optional<PackageDeclaration> jpPackage = cu.getPackageDeclaration();
+        if (jpPackage.isEmpty()) {
+            throw new PackageDeclarationNotFoundException(
+                    "The JavaParser package declaration of the compilation unit is empty"
+            );
         }
-        String methodReturnType = getTypeWithoutPackages(methodUsage.returnType());
-        String methodName = methodUsage.getName();
-        List<String> methodParameters = new ArrayList<>();
-        for (int i=0; i < methodUsage.getNoParams(); i++) {
-            methodParameters.add(getTypeWithoutPackages(methodUsage.getParamType(i)) + " arg" + i);
+        return jpPackage.get();
+    }
+
+    /**
+     * Returns the base element type of a resolved type {@link ResolvedType}.
+     * Recursively strips all array variables. For example:
+     *  Object[][] => Object
+     *
+     * @param resolvedType a type
+     * @return the base component type
+     */
+    public static ResolvedType removeArray(ResolvedType resolvedType) {
+        if (resolvedType.isArray()) {
+            return removeArray(resolvedType.asArrayType().getComponentType());
         }
-        List<String> methodExceptions = new ArrayList<>();
-        for (ResolvedType exceptionType : methodUsage.exceptionTypes()) {
-            methodExceptions.add(getTypeWithoutPackages(exceptionType));
+        return resolvedType;
+    }
+
+    /**
+     * @param resolvedType a JavaParser resolved type
+     * @return true iff a given type is generic. Ignores any wrapped arrays.
+     */
+    public static boolean isGenericType(
+            ResolvedType resolvedType
+    ) {
+        ResolvedType componentType = removeArray(resolvedType);
+        return componentType.isTypeVariable();
+    }
+
+    /**
+     * Returns true iff a given type name represents a generic type.
+     *
+     * @param jpTypeName the name of a type
+     * @param jpCallable a method
```

Comment:
How is this related to the method or used by it?  The documentation doesn't mention it.